src/tools/lib/python/picongpu/plugins/energy_histogram.py

```diff
@@ -0,0 +1,160 @@
+import os
+
+
+class EnergyHistogram(object):
+    """
+    Data Reader for the Energy Histogram Plugin.
+    """
+
+    def __init__(self, run_directory):
+        """
+        Parameters
+        ----------
+        simulation_directory : string
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        if run_directory is None:
+            raise ValueError('The run_directory parameter can not be None!')
+
+        self.run_directory = run_directory
+        self.data_file_suffix = "_energyHistogram.dat"
+
+    def get_data_path(self, species):
+        """
+        Return the path to the underlying data file.
+
+        Parameters
+        ----------
+        species : string
+            short name of the particle species, e.g. 'e' for electrons
+            (defined in ``speciesDefinition.param``)
+
+        Returns
+        -------
+        A string with a path.
+        """
+        if species is None:
+            raise ValueError('The species parameter can not be None!')
+
+        sim_ouput_dir = os.path.join(self.run_directory, "simOutput")
+        if not os.path.isdir(sim_ouput_dir):
+            raise IOError('The simOutput/ directory does not exist inside '
+                          'path:\n  {}\n'
+                          'Did you set the proper path to the run directory?\n'
+                          'Did the simulation already run?'
+                          .format(self.run_directory))
+
+        data_file_path = os.path.join(sim_ouput_dir,
+                                      species + self.data_file_suffix)
+        if not os.path.isfile(data_file_path):
+            raise IOError('The file {} does not exist.\n'
+                          'Did the simulation already run?'
+                          .format(data_file_path))
+
+        return data_file_path
+
+    def get_iterations(self, species):
+        """
+        Return the an array of iterations with available data.
+
+        Parameters
+        ----------
+        species : string
+            short name of the particle species, e.g. 'e' for electrons
+            (defined in ``speciesDefinition.param``)
+
+        Returns
+        -------
+        An array with unsigned integers.
+        """
+        data_file_path = self.get_data_path(species)
+
+        # the first column contains the iterations
+        return np.loadtxt(data_file_path,
+                          usecols=range(0, 1),
+                          dtype=np.uint64)
+
+    def get(self, species, iteration=None, t=None, include_overflow=False):
+        """
+        Get a histogram.
+
+        Parameters
+        ----------
+        species : string
+            short name of the particle species, e.g. 'e' for electrons
+            (defined in ``speciesDefinition.param``)
+        iteration : (unsigned) int [unitless], optional
+            The iteration at which to read the data.
+            Either `t` or `iteration` must be given.
```

Comment:
OPTIONAL:
~~please add a `@TODO` here too~~
Is fine - check is implemented, only `t` does not work.